source/adapters/level_zero/command_buffer.cpp

```diff
@@ -737,18 +731,29 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferEnqueueExp(
     else
       CommandBuffer->WaitEvent->WaitList.insert(TmpWaitList);
 
-    ZE2UR_CALL(zeCommandListAppendBarrier,
-               (WaitCommandList->first, CommandBuffer->WaitEvent->ZeEvent,
-                CommandBuffer->WaitEvent->WaitList.Length,
-                CommandBuffer->WaitEvent->WaitList.ZeEventList));
-  } else {
-    UR_CALL(Queue->Context->getAvailableCommandList(Queue, WaitCommandList,
-                                                    false, false));
+    if (CommandBuffer->WaitEvent->WaitList.Length > 0) {
+      ZE2UR_CALL(zeCommandListAppendBarrier,
+                 (WaitCommandList->first, CommandBuffer->WaitEvent->ZeEvent,
+                  CommandBuffer->WaitEvent->WaitList.Length,
+                  CommandBuffer->WaitEvent->WaitList.ZeEventList));
+      Queue->executeCommandList(WaitCommandList, false, false);
```

Comment:
Think we can move `ur_command_list_ptr_t WaitCommandList{};` declaration to inside this if scope now, can 
also maybe only do ` UR_CALL(Queue->Context->getAvailableCommandList(Queue, WaitCommandList,
false, false))` if wait list isn't empty

Reply:
The scope i was hoping we could move `ur_command_list_ptr_t WaitCommandList{` and 

```
 UR_CALL(Queue->Context->getAvailableCommandList(Queue, WaitCommandList,
                                                    false, false))
```
to is inside this `if (!CommandBuffer->WaitEvent->WaitList.isEmpty()) {`. So we only find an available command-list if we're going to execute it.

Reply:
Yes. Sorry. You are right. I fixed that.